semgrep-core/src/parsing/pfff/ruby_to_generic.ml

```diff
@@ -436,12 +436,18 @@ and literal x =
       | Tick (l, xs, r) ->
           G.OtherExpr
             (("Subshell", l), [ G.E (string_contents_list (l, xs, r) |> G.e) ]))
-  | Regexp ((l, xs, r), opt) -> (
-      match xs with
-      | [ StrChars (s, t) ] -> G.L (G.Regexp ((l, (s, t), r), opt))
-      | _ ->
-          (* TODO *)
-          string_contents_list (l, xs, r))
+  | Regexp ((l, xs, r), opt) ->
+      let rec f strs toks = function
+        | [ StrChars (s, t) ] ->
+            let str = String.concat "" (s :: strs) in
+            let tok = PI.combine_infos t toks in
+            G.L (G.Regexp ((l, (str, tok), r), opt))
+        | StrChars (s, t) :: tl -> f (s :: strs) (t :: toks) tl
+        | _ ->
+            (* TODO *)
+            string_contents_list (l, xs, r)
```

Comment:
Is this typed enough that this can be
```ocaml
| [] -> ...
```
?

Reply:
Are you asking whether `[]` can ever flow into this function? My guess is no, but I'm not sure. Either way, it will have the same result as it previously did.

Reply:
Oh, I think I understand. You're asking whether I can replace the wildcard match with a match against `[]`. No, we could also encounter `StrExpr` instead of `StrChars` at any point in the list.

Reply:
I've updated it to be an exhaustive match without using the wildcard. Should be clearer now.